### PR TITLE
Automerge bot on GCP

### PR DESCRIPTION
**Pull request by balopat**
Automerge bot on GCP

This PR adds the initial information about our GCP infrastructure and also the first service on it: our automerge bot.

### CODE REVIEW

File: dev_tools/auto_merge/README.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
+# Automerge bot 
+
+This automerge but continuously watches a github repository for PRs labelled with the
```
Comment (kevinsung) [inline dev_tools/auto_merge/README.md:3]: but -> bot

Comment (balopat) [reply dev_tools/auto_merge/README.md]: Good catch! Ack.

File: dev_tools/auto_merge/README.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+# Automerge bot 
+
+The automerge bot continuously watches a github repository for PRs labelled with the
+'automerge' label. When it sees such a PR it will mark it by labelling it with
+the 'front_of_queue_automerge' label. While there is an 'front_of_queue_automerge'
+labelled PR, the script will not label any other PRs with 'front_of_queue_automerge'.
+
+While there is a 'front_of_queue_automerge' labelled PR, the script will sync that PR
```
Comment (Strilanc) [inline dev_tools/auto_merge/README.md:8]: It actually picks only one of the PRs to focus on. It prefers PRs that require less work (e.g. tests already passed) and then breaks ties using the lowest PR number.

**Review by Strilanc — APPROVED** [review]

File: dev_tools/auto_merge/README.md (hunk 2 above)
Comment (balopat) [reply dev_tools/auto_merge/README.md]: Thanks for pointing that out - clarified!

**Pull request merged**